submodules/k8s/templates/k8s-functions.sh.tftpl

```diff
@@ -4,17 +4,21 @@ RED="\e[31m"
 GREEN="\e[32m"
 EC="\e[0m"
 
+KUBECONFIG="${kubeconfig_path}"
+KUBECONFIG_PROXY="$KUBECONFIG-proxy"
+
 open_ssh_tunnel_to_k8s_api() {
   TUNNEL_SOCKET_FILE=$${TUNNEL_SOCKET_FILE:-/tmp/k8s-tunnel-socket}
   printf "$GREEN Openning k8s tunnel ... $EC \n"
   ssh -q -N -f -M -o "StrictHostKeyChecking no" -o "ExitOnForwardFailure=yes" -i "${ssh_pvt_key_path}" -D ${k8s_tunnel_port} -S "$TUNNEL_SOCKET_FILE" ${bastion_user}@${bastion_public_ip}
 }
 
 check_kubeconfig() {
-  local KUBECONFIG="${kubeconfig_path}"
   printf "$GREEN Checking if $KUBECONFIG exists... $EC \n"
   if test -f "$KUBECONFIG"; then
-    echo "$KUBECONFIG exists." && export KUBECONFIG
+    echo "$KUBECONFIG exists, creating $KUBECONFIG_PROXY for proxy use."
+    cp $KUBECONFIG $KUBECONFIG_PROXY
+    kubectl --kubeconfig $KUBECONFIG_PROXY config set "clusters.${eks_cluster_arn}.proxy-url" "socks5://127.0.0.1:${k8s_tunnel_port}"
```

Comment:
you could also just use `kubectl set-cluster <cluster name> --proxy-url=....`

Reply:
This only works in kubectls 1.24 and greater, so we'll break 1.23 and earlier provisioning while it's still possible and reduce compatibility on places with a non-current version of kubectl that would have worked if we'd used the more compatible command.

Reply:
actually we pick the name that we are configuring with.
look at eks/cluster.tf.  We can output the cluster name.

Reply:
I think you might actually be responding to the other comment? I'm going to assume so in my response.

If you mean the `--alias` argument, afaict that only impacts the name of the context. We provide it that argument, and the context gets named the plaintext name of the clsuter, the actual cluster name in the kubeconfig (which we need to reference here) is still the cluster's arn:

```
- cluster:
    certificate-authority-data: <sanitized>
    # proxy-url needs to go here, which is keyed by the arn in `cluster[0].name` below
    server: https://sanitized.gr7.us-west-2.eks.amazonaws.com
  name: arn:aws:eks:us-west-2:1234567890:cluster/chibipug6680
contexts:
- context:
    cluster: arn:aws:eks:us-west-2:1234567890:cluster/chibipug6680
    user: arn:aws:eks:us-west-2:1234567890:cluster/chibipug6680
  name: chibipug6680
current-context: chibipug6680
```

If you mean the `aws_cluster.this.name` attribute, then it's just the incorrect attribute for what we need to target in the kubeconfig

Reply:
no, just need the cluster arn to be returned. I guess you could return the entire cluster and then just pull the arn as well.

Reply:
never mind... its what you are doing.  the docs for update-kubeconfig use incorrect wording but in the end it doesn't matter since we are doing the right thing.